pallets/automation-time/src/types.rs

```diff
@@ -185,3 +202,114 @@ impl<AccountId> TaskHashInput<AccountId> {
 		Self { owner_id, provided_id }
 	}
 }
+
+#[derive(Debug, Decode, Eq, Encode, PartialEq, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct ScheduledTasks<AccountId, TaskId> {
+	pub tasks: Vec<(AccountId, TaskId)>,
+	pub weight: u128,
+}
+impl<A, B> Default for ScheduledTasks<A, B> {
+	fn default() -> Self {
+		Self { tasks: vec![], weight: 0 }
+	}
+}
+impl<AccountId, TaskId> ScheduledTasks<AccountId, TaskId> {
+	pub fn try_push<T: Config, Balance: AtLeast32BitUnsigned>(
+		&mut self,
+		task_id: TaskId,
+		task: &Task<AccountId, Balance, T::CurrencyId, T::MaxExecutionTimes>,
+	) -> Result<&mut Self, Error<T>>
+	where
+		AccountId: Clone,
+	{
+		let weight = self
+			.weight
+			.checked_add(task.action.execution_weight::<T>() as u128)
+			.ok_or(Error::<T>::TimeSlotFull)?;
+		if self.tasks.len() as u32 >= T::MaxTasksPerSlot::get() ||
+			weight > T::MaxWeightPerSlot::get()
```

Comment:
isn't this condition met already in the check above?

Reply:
```
.checked_add(task.action.execution_weight::<T>() as u128)
.ok_or(Error::<T>::TimeSlotFull)?;
```
only guards against overflow of the u128 type. This should never actually happen but since saturating could have unintended consequences on `>=` checks I opted to explicitly check the error
```
self.tasks.len() as u32 >= T::MaxTasksPerSlot::get()
```
Limits some of the smaller tasks but is not sufficient for heavier tasks like auto compounding (and in the future dynamic actions)